### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by sixdouglas**
Breaking changes.
Moving from raw ArrayList typed List<>. 
Adding TableRectangle class as parent of Table and Cell in order to enforce the restriction that where only in the JavaDoc.

### CODE REVIEW

**Comment by sixdouglas** [comment]
Linked to issue #193

File: pdf-toolbox/src/test/java/com/lowagie/examples/RunAll.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import com.lowagie.examples.directcontent.optionalcontent.OptionalContent;
+import com.lowagie.examples.directcontent.optionalcontent.OrderedLayers;
+import com.lowagie.examples.directcontent.pageevents.Bookmarks;
+import com.lowagie.examples.directcontent.pageevents.EndPage;
+import com.lowagie.examples.directcontent.pageevents.Events;
+import com.lowagie.examples.directcontent.pageevents.PageNumbersWatermark;
+import com.lowagie.examples.directcontent.text.Logo;
+import com.lowagie.examples.directcontent.text.Text;
+
+public class RunAll {
```
Comment (sixdouglas) [inline pdf-toolbox/src/test/java/com/lowagie/examples/RunAll.java]: Adding this one to run them all and simplify the testing

**Comment by andreasrosdal** [comment]
Please add highlevel description.

**Comment by andreasrosdal** [comment]
Does this change break the API for existing users?

**Pull request merged**